deeprankcore/trainer.py

```diff
@@ -202,23 +235,53 @@ def _load_model(self):
         self.configure_optimizers()
         self.set_loss()
 
-    def _check_dataset_equivalence(self):
-        for other in [self.dataset_val, self.dataset_test]:
-            if other is not None:
-                if (other.target == self.target # pylint: disable = too-many-boolean-expressions
-                    and other.node_features == self.node_features
-                    and other.edge_features == self.edge_features
-                    and other.clustering_method == self.clustering_method
-                    and other.task == self.task
-                    and other.classes == self.classes
-                    ):
-                    pass
-                else:
-                    raise ValueError(
-                        f"""Training dataset is not equivalent to {other}.\n
-                        Check datasets passed to Trainer class and ensure that the same (non-default) \n
-                        target, node_features, edge_features, clustering_method, task, and classes are used."""
+    def _check_dataset_equivalence(self, train_dataset, validation_dataset, test_dataset):
```

Comment:
```suggestion
    def _check_dataset_equivalence(self, dataset_train, dataset_val, dataset_test):
```

Reply:
Changed this in commit 9ad784c964ef77876f71e39599c150dcace6bb22.